# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1311570686 25200
#      Sun Jul 24 22:11:26 2011 -0700
# Node ID ff264eae88cf46a2b4192f806641b6d5bcc5afc2
# Parent  c3b305e7e59fc6096d4c09663f57730b254e284d
textread.m, textscan.m: Read entire file when format repeat count is -1

* textread.m, textscan.m: Read entire file when format repeat count is -1.

diff --git a/scripts/io/textread.m b/scripts/io/textread.m
--- a/scripts/io/textread.m
+++ b/scripts/io/textread.m
@@ -73,17 +73,17 @@ function varargout = textread (filename,
   endif
 
   if (nargin > 2 && isnumeric (varargin{1}))
     nlines = varargin{1};
   else
     nlines = Inf;
   endif
 
-  if (isfinite (nlines))
+  if (isfinite (nlines) && (nlines >= 0))
     str = tmp_str = "";
     n = 0;
     ## FIXME: Can this be done without slow loop?
     while (ischar (tmp_str) && n++ <= nlines)
       str = strcat (str, tmp_str);
       tmp_str = fgets (fid);
     endwhile
   else
diff --git a/scripts/io/textscan.m b/scripts/io/textscan.m
--- a/scripts/io/textscan.m
+++ b/scripts/io/textscan.m
@@ -134,17 +134,17 @@ function [C, position] = textscan (fid, 
   else
     ## Skip header lines if requested
     headerlines = find (strcmpi (args, "headerlines"), 1);
     ## Beware of zero valued headerline, fskipl would skip to EOF
     if (! isempty (headerlines) && (args{headerlines + 1} > 0))
       fskipl (fid, varargin{headerlines + 1});
       args(headerlines:headerlines+1) = []; 
     endif
-    if (isfinite (nlines))
+    if (isfinite (nlines) && (nlines >= 0))
       str = tmp_str = "";
       n = 0;
       ## FIXME: Can this be done without slow loop?
       while (ischar (tmp_str) && n++ <= nlines)
         str = strcat (str, tmp_str);
         tmp_str = fgets (fid);
       endwhile
     else
